src/board/Board.jsx

```diff
@@ -4,13 +4,15 @@ import "./Board.css";
 import { updateBoard } from "./updateBoard";
 import { defaultBoardData } from "../board/boardData";
 import PropTypes from "prop-types";
+import { getBoardState } from "./getBoardState";
 
-function Board({ changeTurn, currentPlayer }) {
+function Board({ changeTurn, currentPlayer, changeState }) {
   const [boardData, setBoardData] = useState(defaultBoardData);
 
   const handleBoardClick = (cellSpot) => {
     const board = updateBoard(boardData, cellSpot, currentPlayer);
```

Comment:
I'm a little confused about this logic. Inside #updateBoard there's only 1 if check. 
```
export function updateBoard(board, cellSpot, currentPlayer) {
  if (
    cellSpot === "X" ||
    cellSpot === "O" ||
    getBoardState(board) !== "Ongoing"
  ) {
    return board;
  } 
```

if it doesn't meet that `if` condition, then `board`  on line 13 here is undefined and you'd be sending setBoardData(undefined).  is that intentional?

Reply:
hmm. I am not sure I understand your question. 

The `updateBoard` function receives a `board`, `cellSpot`, and `currentPlayer` as input. It checks `if` the `cellSpot` is already occupied by "X" or "O" or `if` the game state is not "Ongoing". If any of these conditions are true, it returns the original `board` unchanged. Otherwise, it returns a new board by updating the specified `cellSpot` with the `currentPlayer`. In any case, the function doesn't return undefined and always provides either the original board or a modified version with the appropriate player's mark placed at the specified spot. Also, if this was returning an undefined, every time a user made a valid move, wouldn't the board not update but error out?

Are you saying there should be more error handling for other edge cases?

Reply:
sorry, that was my bad!  github didn't show me the lines below that if block and I got confused.  feel free to ignore my comment!